src/Illuminate/Database/Connection.php

```diff
@@ -502,11 +502,13 @@ public function prepareBindings(array $bindings)
      * Execute a Closure within a transaction.
      *
      * @param  \Closure  $callback
+     * @param  callable|null  $success
+     * @param  callable|null  $error
      * @return mixed
      *
      * @throws \Exception|\Throwable
      */
-    public function transaction(Closure $callback)
+    public function transaction(Closure $callback, callable $success = null, callable $error = null)
```

Comment:
What will be better? Change first to `callable` or new to `Closure`? How do you think?